### PR TITLE
fix: allow projected pre rendered code examples

### PR DESCRIPTION
**Pull request by coryrylan**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] If applicable, have a visual design approval

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] clarity.design website / infrastructure changes
- [ ] Other... Please describe:

## What is the new behavior?
This allows code demos to be imported and rendered at build time for the website. This improves SEO as the code samples are all pre-rendered into the html without the need of JS or PrismJS runtime. This also improves performance as Prism does not have to run for each code block. 

I only updated one doc page for this PR. This can wait till after the website release. I can help out with other pages in follow up PRs as well.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: apps/website/components/button/README.md [hunk 1]
```diff
@@ -219,39 +219,84 @@ If your icon button is has no text, we recommend adding the `title="` attribute
 
 ### Solid Buttons
 
-<doc-demo src="/demos/button/solid-css.html" demo="/demos/button/solid-css.html"></doc-demo>
+<doc-demo>
+<<< .vuepress/public/demos/button/solid-css.html
+::: slot demo
+!!!include(.vuepress/public/demos/button/solid-css.html)!!!
+:::
```
Comment (coryrylan) [inline apps/website/components/button/README.md:226]: The ```<<<``` imports the file as a code snippet and is pre-rendered by vuepress at build time.

The `::: slot demo``` tells vuepress to render the code in the `code` slot of the component.

The `!!!include()``` imports and inlines the html into the slot rendering the html at build time. Kind of like a Jekyll includes.

**Comment by gnomeontherun** [comment]
This works for some of the demos, but can't work for TS examples. Before we change to this, I want to make sure we are clear on what we'll optimize and what we might not.

**Comment by mathisscott** [comment]
@coryrylan 
Please rebase and merge for v4.

**Comment by gnomeontherun** [comment]
I'd like to see us pull this through fully, which might be a separate task and not on Cory to do. I'm leaving this open as guidance and it will help us scope the work.

**Comment by gnomeontherun** [comment]
I'm pulling this through in a new PR, so closing this one. Thanks Cory for the ideation and I've improved upon this slightly.

**Pull request closed without merging**